src/equisolve/numpy/models/linear_model.py

```diff
@@ -53,19 +57,16 @@ def _validate_data(self, X: TensorMap, y: Optional[TensorMap] = None) -> None:
         :param y:
             target data to check
         """
-        if len(X.components_names) != 0:
-            raise ValueError("`X` contains components")
-
         if y is not None:
             _check_maps(X, y, "_validate_data")
 
-            if len(y.components_names) != 0:
-                raise ValueError("`y` contains components")
-
             for key, X_block in X:
                 y_block = y.block(key)
                 _check_blocks(
-                    X_block, y_block, props=["samples"], fname="_validate_data"
+                    X_block,
+                    y_block,
+                    props=["samples", "components"],
+                    fname="_validate_data",
                 )
 
                 if len(y_block.properties) != 1:
```

Comment:
I think we can get rid of this condition because we don't necessarily need the length of block.properties to be 1, we can be learning multiple things. For example in the electron density learning project, the target has the "radial_channels" in the properties (as it tries to learn the coefficients of different orbital basis).

Reply:
If this works with `scipy.linalg.solve` and `scipy.linalg.lstsq` I am happy to allow this.